week05/errors_test.go

```diff
@@ -0,0 +1,92 @@
+package demo
+
+import (
+	"fmt"
+	"testing"
+)
+
+func Test_tablenotfound(t *testing.T) {
+	t.Parallel()
+
+	tn := "users"
+
+	errTable := ErrTableNotFound{
+		table: tn,
+	}
+
+	if errTable.Error() == "" {
+		t.Fatalf("table %s not found", errTable.table)
+	}
+
+}
+
+func Test_Table_Errors2(t *testing.T) {
+	t.Parallel()
+
+	s := &Store{}
+	tn := "users"
+
+	s.Insert(tn, Model{})
+
+	errTable := ErrTableNotFound{
+		table: tn,
+	}
+
+	if !IsErrTableNotFound(&errTable) {
+		t.Fatalf("table not found %v", errTable.table)
```

Comment:
This error message doesn't tell me what happened, in fact its very confusing in the context of this text.

```suggestion
		t.Fatalf("expected %T to be an ErrTableNotFound", errTable)
```